src/ping.rs

```diff
@@ -0,0 +1,86 @@
+//! Module containing the `/ping` application command.
+
+use crate::ApplicationCommandHandler;
+
+use serenity::async_trait;
+
+use serenity::model::guild::Guild;
+use serenity::model::id::GuildId;
+use serenity::model::interactions::application_command::ApplicationCommandInteraction;
+use serenity::prelude::*;
+
+use std::collections::HashSet;
+use std::error::Error;
+
+/// Keeps track of guilds for which this application command handler has already
+/// been registered. When this application command is registered to a new guild,
+/// the corresponding [`GuildId`] is added to [`PingRegisteredInfo::Value`]
+#[doc(hidden)]
+struct PingRegisteredInfo;
+
+impl TypeMapKey for PingRegisteredInfo {
+    type Value = HashSet<GuildId>;
+}
+
+/// A basic [`ApplicationCommandHandler`], intended for demonstration purposes.
+///
+/// This [`ApplicationCommandHandler`] simply replies to `/ping` with
+/// `Pong, <user>!`.
+pub struct PingApplicationCommand;
+
+#[async_trait]
+impl ApplicationCommandHandler for PingApplicationCommand {
+    async fn register_command(
```

Comment:
is there a potential DRY-violation in re-implementing `register_command` for every command?

Reply:
How were you thinking of refactoring this logic out? Aside from registering commands with different options, different implementations may store various objects in the global context, in addition to a variety of other checks.

Reply:
That makes sense to me. LGTM